cicecore/cicedynB/infrastructure/ice_read_write.F90

```diff
@@ -68,6 +68,9 @@ module ice_read_write
                          ice_read_nc_xyz, &
                          !ice_read_nc_xyf, &
                          ice_read_nc_point, &
+                         ice_read_nc_1D,  &
+                         ice_read_nc_2D,  &
```

Comment:
What distinguishes ice_read_nc_2D from _xy?  similarly for 1D/point, 3D/xyz, etc.  Does it make sense to combine any of them?  I also see ice_read_vec_nc, and _uv, and _xyf, above...  These routines are multiplying!  I'm mostly just curious, not asserting that anything has to be done about it at the moment.

Reply:
I wanted to use the other implementations, but they are built for CICE arrays specifically and tend to assume sizes and halos.  They are not for general 1D, 2D, 3D arrays, so I had to add those.